### ISSUE TITLE
how to training the "little" model 

### ISSUE THREAD

**Issue by battlefor** [xiph/rnnoise#246]
hello,I glad to obtain the amazing rnnoise lib,it is very suitable for me.

now i want to training the "little" model ,just like the "little" model  provided with half as an alternative

i had study the  rnnoise_data_little.c for  a long time,  but i still don't know how to get the rnnoise_data_little.c from my train.

**Comment by BancoLin** [comment]
@battlefor The rnnoise training script already covers 'little model' training steps (model sparsification), but your training data must large enough to trigger it. If I recall correctly, the little model training steps start at iteration 2500 and stop at iteration 8000. You can check your training log to see if the training runs beyond 2500 iterations. 

If you want to fine tune the sparsification parameters, please go to check `rnnoise/rnnoise.py`. There are `sparsify_start`, `sparsify_stop`, `sparsify_interval`, and `sparsify_exponent`. You can control `sparsify_start` parameter to control when to start model sparsification.

**Comment by Meonardo** [comment]
> the little model training steps start at iteration 2500 and stop at iteration 8000

the `iteration`, do you mean in one epoch or the whole training process?

**Comment by BancoLin** [comment]
> > the little model training steps start at iteration 2500 and stop at iteration 8000
> 
> the `iteration`, do you mean in one epoch or the whole training process?

the whole training process.

the total number of iterations = #epoch * #training_steps_per_epoch

**Comment by Meonardo** [comment]
@BancoLin thanks for reply, I have managed to run `200` epochs for my training, each epoch takes `3333` steps, my `sparsify_start` & `sparsify_stop` left to default values which are: `6000` & `20000`.

so this is mean I can dump the `rnnoise_data_little.c` use checkpoint file `rnnoise_7.pth`?

**Comment by BancoLin** [comment]
> [@BancoLin](https://github.com/BancoLin) thanks for reply, I have managed to run `200` epochs for my training, each epoch takes `3333` steps, my `sparsify_start` & `sparsify_stop` left to default values which are: `6000` & `20000`.
> 
> so this is mean I can dump the `rnnoise_data_little.c` use checkpoint file `rnnoise_7.pth`?

in theory, yes
in practice, you should first check whether training loss is stabilized or not.

once you dump the checkpoint file, please check `rnnoise_data.c` to see  dimensions are reduced as expected.
For example, the original dimension of `gru1_input_weights_float` is 147456
```
static const float gru1_input_weights_float[147456]
```

After sparsification, the dimension reduces to 58976
```
static const float gru1_input_weights_float[58976]
```
